lib/interpreters/periodic_visits.rb

```diff
@@ -111,47 +120,45 @@ def generate_relations_between_visits(vrp, mission)
       # TODO : need to uniformize generated relations whether mission has minimum AND maximum lapse or only one of them
       return unless mission.visits_number > 1
 
-      if mission.minimum_lapse && mission.maximum_lapse
-        (2..mission.visits_number).each{ |index|
+      if mission.minimum_lapse && mission.maximum_lapse && (mission.minimum_lapse == mission.maximum_lapse)
+        2.upto(mission.visits_number){ |index|
           current_lapse = (index - 1) * mission.minimum_lapse.to_i
           vrp.relations << Models::Relation.create(
             type: :minimum_day_lapse,
             linked_ids: ["#{mission.id}_1_#{mission.visits_number}", "#{mission.id}_#{index}_#{mission.visits_number}"],
             lapses: [current_lapse]
           )
         }
-        (2..mission.visits_number).each{ |index|
+        2.upto(mission.visits_number){ |index|
           current_lapse = (index - 1) * mission.maximum_lapse.to_i
           vrp.relations << Models::Relation.create(
             type: :maximum_day_lapse,
             linked_ids: ["#{mission.id}_1_#{mission.visits_number}", "#{mission.id}_#{index}_#{mission.visits_number}"],
             lapses: [current_lapse]
           )
         }
-      elsif mission.minimum_lapse
-        (2..mission.visits_number).each{ |index|
-          current_lapse = mission.minimum_lapse.to_i
+      else
+        if mission.minimum_lapse
           vrp.relations << Models::Relation.create(
             type: :minimum_day_lapse,
-            linked_ids: ["#{mission.id}_#{index - 1}_#{mission.visits_number}", "#{mission.id}_#{index}_#{mission.visits_number}"],
-            lapses: [current_lapse]
+            linked_ids: 1.upto(mission.visits_number).map{ |index| "#{mission.id}_#{index}_#{mission.visits_number}" },
+            lapses: [mission.minimum_lapse.to_i]
```

Comment:
I don't know, I copied the exact logic -- there was `current_lapse = mission.minimum_lapse.to_i`. I suspect it can be `nil` but not sure.

Reply:
Resolving this, you can reopen if not satisfied with the answer.

Reply:
The condition to reach this code is `if mission.minimum_lapse` so I doubt it can be nil here :slightly_smiling_face:

Reply:
I didn't dig deeper since it wasn't the purpose of the simplification and I kept the exact the same functioning but if you have checked and sure that's the case, then I can remove the `to_i`. 

However, I don't see why we should take a risk here, since `integer.to_i` is basically free (and I am not sure if we allow non-integer lapses :shrug:)...

Reply:
In the `vrp_input.rb` they  both are defined as `Float` so I think it is better to leave it as it is. What do you think?

Reply:
Indeed, you can let it with .to_i :+1: